src/Kestrel.Core/Internal/Http/Http1MessageBody.cs

```diff
@@ -93,12 +93,13 @@ protected Http1MessageBody(Http1Connection context)
                             BadHttpRequestException.Throw(RequestRejectionReason.UnexpectedEndOfRequestContent);
                         }
 
-                        awaitable = _context.Input.ReadAsync();
                     }
                     finally
                     {
                         _context.Input.AdvanceTo(consumed, examined);
                     }
+
+                    awaitable = _context.Input.ReadAsync();
```

Comment:
Why did this move?

Reply:
Because in sync case `ReadAsync` would be called before `AdvanceTo` and throw